File: pset6/mario/more/mario.py
from cs50 import get_int

# Check input and reprompt if not in range
while True:
    height = get_int("Height: ")
    if height > 0 and height < 9:
        break

# Initialize list to iterate over
hlist = [i for i in range(1, height + 1)]

# printing hashes and spaces
for i in hlist:
    print(" " * (height - i), end="")
    print("#" * i, end="")
    print("  ", end="")
    print("#" * i)

File: pset1/mario.c
#include <cs50.h>
#include <stdio.h>


void print_spaces (int n);
void print_hashes (int n);

int main (void)
{
    int h;
   // Get a positive int from user between 0-23
    do
    {
       h = get_int("Height: ");
    }
    while (h < 0 || h > 23);

    // iteration for printing each line of the pyramid
    for (int i = 0; i < h; i++)
    {

       print_spaces(h-1-i); //print initial spaces in line i (left side)
       print_hashes(i+1); //print hashes in line i (left side)
       print_spaces(2); // print gap between the pyramid sides
       print_hashes(i+1); // print hashes on the right side
       printf("\n");
    }
}

// function for printing "n" spaces
void print_spaces (int n)
{
    for (int i = 0; i < n; i++)
    {
        printf(" ");
    }
}

// function for printing "n" hashes
void print_hashes (int n)
{
    for (int i = 0; i < n; i++)
    {
        printf("#");
    }
}